### ISSUE TITLE
fishtest down?

### ISSUE THREAD

**Issue by vondele** [glinscott/fishtest#403]
current main page of fishtest leads to an internal error. I can still access a specific user page, but login is not possible.

**Comment by 31m059** [comment]
Most recent user action was 19-09-27 22:39:23, 8-9 hours ago.  So it's been down for quite some time.

However, as often is the case when fishtest seems "down", the new test page and actions page still appear to work:
http://tests.stockfishchess.org/tests/run
http://tests.stockfishchess.org/actions
Other miscellaneous pages, including individual tests and live Elo still work.

So we can continue to use fishtest in this state (as long as you're already logged in) by using the above two links.  Approvers will need to find which tests have been submitted on the Actions page, then click the test to inspect and approve.

**Comment by 31m059** [comment]
I can confirm: tests can still be submitted from the above URL.
http://tests.stockfishchess.org/tests/view/5d8f0eba0ebc590f3bebdde6

**Comment by vondele** [comment]
I'm not logged in, so I can't check...

**Comment by 31m059** [comment]
Test is approved and workers are allocated.  So everything is working behind the scenes, except the main landing page.

**Comment by vondele** [comment]
and login page :-)

**Comment by vondele** [comment]
actually, it seems like I did login, even though the internal error page appeared on login.

**Comment by 31m059** [comment]
@vondele That makes sense, since a login redirects you to the main page (which is the part that's broken).  I was wondering how you approved my test if you weren't logged in :)

**Comment by tomtor** [comment]
Shiwing Green and slow tests also work. The last action was a test with failed signature, so for some reason that test is not showable. We will have to wait till @ppigazzini can display the python stack trace.

**Comment by tomtor** [comment]
http://tests.stockfishchess.org/tests?page=2 works also

**Comment by ppigazzini** [comment]
```
2019-09-28 10:13:57,932 ERROR [waitress][Dummy-10] Exception when serving /tests
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/waitress/channel.py", line 332, in service
    task.service()
  File "/usr/lib/python2.6/site-packages/waitress/task.py", line 174, in service
    self.execute()
  File "/usr/lib/python2.6/site-packages/waitress/task.py", line 382, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 251, in __call__
    response = self.invoke_subrequest(request, use_tweens=True)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 227, in invoke_subrequest
    response = handle_request(request)
  File "/usr/lib/python2.6/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 161, in handle_request
    response = view_callable(context, request)
  File "/usr/lib/python2.6/site-packages/pyramid/config/views.py", line 369, in rendered_view
    context)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 531, in render_view
    return self.render_to_response(response, system, request=request)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 561, in render_to_response
    result = self.render(value, system_values, request=request)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 557, in render
    result = renderer(value, system_values)
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 240, in __call__
    reraise(MakoRenderingException(errtext), None, exc_info[2])
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 232, in __call__
    result = template.render_unicode(**system)
  File "/usr/lib/python2.6/site-packages/mako/template.py", line 452, in render_unicode
    as_unicode=True)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 783, in _render
    **_kwargs_for_callable(callable_, data))
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 815, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 841, in _exec_template
    callable_(context, *args, **kwargs)
  File "base_mak", line 42, in render_body
  File "tests_mak", line 69, in render_body
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "run_table_mak", line 97, in render_body
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "elo_results_mak", line 36, in render_body
MakoRenderingException:

Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 232, in __call__
    result = template.render_unicode(**system)
  File "/usr/lib/python2.6/site-packages/mako/template.py", line 452, in render_unicode
    as_unicode=True)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 783, in _render
    **_kwargs_for_callable(callable_, data))
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 815, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 841, in _exec_template
    callable_(context, *args, **kwargs)
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/base.mak", line 83, in render_body
    ${self.body()}
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/tests.mak", line 20, in render_body
    <%include file="run_table.mak" args="runs=runs['failed'], show_delete=True"/>
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/run_table.mak", line 71, in render_body
    <td style="width:1%"><%include file="elo_results.mak" args="run=run, show_gauge=active" /></td>
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/elo_results.mak", line 10, in render_body
    %if 'sprt' in run['args'] and not 'Pending' in run['results_info']['info'][0]:
KeyError: 'results_info'

2019-09-28 10:14:28,180 ERROR [waitress][Dummy-10] Exception when serving /tests
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/waitress/channel.py", line 332, in service
    task.service()
  File "/usr/lib/python2.6/site-packages/waitress/task.py", line 174, in service
    self.execute()
  File "/usr/lib/python2.6/site-packages/waitress/task.py", line 382, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 251, in __call__
    response = self.invoke_subrequest(request, use_tweens=True)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 227, in invoke_subrequest
    response = handle_request(request)
  File "/usr/lib/python2.6/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 161, in handle_request
    response = view_callable(context, request)
  File "/usr/lib/python2.6/site-packages/pyramid/config/views.py", line 369, in rendered_view
    context)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 531, in render_view
    return self.render_to_response(response, system, request=request)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 561, in render_to_response
    result = self.render(value, system_values, request=request)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 557, in render
    result = renderer(value, system_values)
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 240, in __call__
    reraise(MakoRenderingException(errtext), None, exc_info[2])
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 232, in __call__
    result = template.render_unicode(**system)
  File "/usr/lib/python2.6/site-packages/mako/template.py", line 452, in render_unicode
    as_unicode=True)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 783, in _render
    **_kwargs_for_callable(callable_, data))
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 815, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 841, in _exec_template
    callable_(context, *args, **kwargs)
  File "base_mak", line 42, in render_body
  File "tests_mak", line 69, in render_body
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "run_table_mak", line 97, in render_body
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "elo_results_mak", line 36, in render_body
MakoRenderingException:

Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 232, in __call__
    result = template.render_unicode(**system)
  File "/usr/lib/python2.6/site-packages/mako/template.py", line 452, in render_unicode
    as_unicode=True)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 783, in _render
    **_kwargs_for_callable(callable_, data))
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 815, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 841, in _exec_template
    callable_(context, *args, **kwargs)
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/base.mak", line 83, in render_body
    ${self.body()}
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/tests.mak", line 20, in render_body
    <%include file="run_table.mak" args="runs=runs['failed'], show_delete=True"/>
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/run_table.mak", line 71, in render_body
    <td style="width:1%"><%include file="elo_results.mak" args="run=run, show_gauge=active" /></td>
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/elo_results.mak", line 10, in render_body
    %if 'sprt' in run['args'] and not 'Pending' in run['results_info']['info'][0]:
KeyError: 'results_info'

2019-09-28 10:14:38,513 ERROR [waitress][Dummy-15] Exception when serving /api/upload_pgn
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/waitress/channel.py", line 332, in service
    task.service()
  File "/usr/lib/python2.6/site-packages/waitress/task.py", line 174, in service
    self.execute()
  File "/usr/lib/python2.6/site-packages/waitress/task.py", line 382, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 251, in __call__
    response = self.invoke_subrequest(request, use_tweens=True)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 227, in invoke_subrequest
    response = handle_request(request)
  File "/usr/lib/python2.6/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 161, in handle_request
    response = view_callable(context, request)
  File "/usr/lib/python2.6/site-packages/pyramid/config/views.py", line 347, in rendered_view
    result = view(context, request)
  File "/usr/lib/python2.6/site-packages/pyramid/config/views.py", line 493, in _requestonly_view
    response = view(request)
  File "/home/fishtest/fishtest/fishtest/fishtest/api.py", line 135, in upload_pgn
    pgn_zip=base64.b64decode(request.json_body['pgn'])
  File "/home/fishtest/fishtest/fishtest/fishtest/rundb.py", line 455, in upload_pgn
    self.pgndb.insert({'run_id': run_id, 'pgn_zip': Binary(pgn_zip)})
  File "/usr/lib64/python2.6/site-packages/pymongo/collection.py", line 357, in insert
    continue_on_error, self.__uuid_subtype), safe)
  File "/usr/lib64/python2.6/site-packages/pymongo/mongo_client.py", line 920, in _send_message
    rv = self.__check_response_to_last_error(response)
  File "/usr/lib64/python2.6/site-packages/pymongo/mongo_client.py", line 863, in __check_response_to_last_error
    raise OperationFailure(details["err"], details["code"])
OperationFailure: document is larger than capped size 220424 > 53248
2019-09-28 10:14:48,265 ERROR [waitress][Dummy-22] Exception when serving /tests
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/waitress/channel.py", line 332, in service
    task.service()
  File "/usr/lib/python2.6/site-packages/waitress/task.py", line 174, in service
    self.execute()
  File "/usr/lib/python2.6/site-packages/waitress/task.py", line 382, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 251, in __call__
    response = self.invoke_subrequest(request, use_tweens=True)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 227, in invoke_subrequest
    response = handle_request(request)
  File "/usr/lib/python2.6/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/usr/lib/python2.6/site-packages/pyramid/router.py", line 161, in handle_request
    response = view_callable(context, request)
  File "/usr/lib/python2.6/site-packages/pyramid/config/views.py", line 369, in rendered_view
    context)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 531, in render_view
    return self.render_to_response(response, system, request=request)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 561, in render_to_response
    result = self.render(value, system_values, request=request)
  File "/usr/lib/python2.6/site-packages/pyramid/renderers.py", line 557, in render
    result = renderer(value, system_values)
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 240, in __call__
    reraise(MakoRenderingException(errtext), None, exc_info[2])
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 232, in __call__
    result = template.render_unicode(**system)
  File "/usr/lib/python2.6/site-packages/mako/template.py", line 452, in render_unicode
    as_unicode=True)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 783, in _render
    **_kwargs_for_callable(callable_, data))
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 815, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 841, in _exec_template
    callable_(context, *args, **kwargs)
  File "base_mak", line 42, in render_body
  File "tests_mak", line 69, in render_body
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "run_table_mak", line 97, in render_body
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "elo_results_mak", line 36, in render_body
MakoRenderingException:

Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/pyramid/mako_templating.py", line 232, in __call__
    result = template.render_unicode(**system)
  File "/usr/lib/python2.6/site-packages/mako/template.py", line 452, in render_unicode
    as_unicode=True)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 783, in _render
    **_kwargs_for_callable(callable_, data))
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 815, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 841, in _exec_template
    callable_(context, *args, **kwargs)
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/base.mak", line 83, in render_body
    ${self.body()}
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/tests.mak", line 20, in render_body
    <%include file="run_table.mak" args="runs=runs['failed'], show_delete=True"/>
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/run_table.mak", line 71, in render_body
    <td style="width:1%"><%include file="elo_results.mak" args="run=run, show_gauge=active" /></td>
  File "/usr/lib/python2.6/site-packages/mako/runtime.py", line 710, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "/home/fishtest/fishtest/fishtest/fishtest/templates/elo_results.mak", line 10, in render_body
    %if 'sprt' in run['args'] and not 'Pending' in run['results_info']['info'][0]:
KeyError: 'results_info'

```

**Comment by tomtor** [comment]
@ppigazzini Could you try https://github.com/glinscott/fishtest/pull/404

**Comment by tomtor** [comment]
@ppigazzini Oops, made a typo. I pushed again...

**Comment by ppigazzini** [comment]
@tomtor changed by hand, it works :)

**Comment by tomtor** [comment]
The trouble with the /tests page is that it not only shows tests but also updates a test when it is first viewed. That is a bit of a hack and should be fixed some day.

**Comment by vondele** [comment]
closing as fixed with #404
